### PR TITLE
Migrate to GraphQL and MongoDB

### PR DESCRIPTION
**Pull request by orkosinha**
This PR adds some basic functionality for migrating the backend to GraphQL and MongoDB.

### Overview
Creates GraphQL based backend for storing and accessing Cornell publications through a MongoDB database.
Articles are queryable by an article ID, by publisher, by a timestamp (certain number of articles published after a timestamp).
Publications are queryable by the publication slug stored in `publications.json` or can be fetched all at once.
Mutations are also available for liking articles (will be changed later) and refreshed (uses `db/rss-parser.ts`).

### Specifics
* Adds resolvers for Articles and Publications `resolvers/*`
* Adds Publication class that is used for GraphQL schema definition `entities/Publication.ts`
* Adds Article class that works as MongoDB model and GraphQL schema definition `entities/Article.ts`
* Adds function for parsing publication feeds and creating Article models `db/rss-parser.ts`
* Adds connection to MongoDB database `db/DBConnection.ts`
* Migrates linting to using `eslint-config-airbnb-typescript-prettier`

Tested using GraphQL playground and MongoDB Compass.

Thanks @manishsaha for figuring out how to fix the linting issue with React.

-- Sorry this PR is extremely fat, a lot of the code is boilerplating for GraphQL and MongoDB apologies

### CODE REVIEW

File: src/db/rss-parser.ts
Comment (grebneerg) [inline src/db/rss-parser.ts:21]: Mostly stylistic suggestion: if you are using node >= 11 (current lts is 14), you could use [Array.prototype.flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) on the result of your nested maps instead of continuously mutating the `articles` variable (the `const` declaration of which is misleading if you ask me).

**Review by grebneerg — APPROVED** [review]
You might want to think about adding some tests for this, but otherwise it looks good!

File: src/entities/Article.ts
Comment (cswenberg) [inline src/entities/Article.ts:15]: i've never done a TS backend project so maybe I just don't know the syntax, but shouldn't this be a FK to reference the other Publication entity?

File: src/entities/Publication.ts
Comment (cswenberg) [inline src/entities/Publication.ts:15]: inconsistent naming with `imageURL` from `Article.ts`

File: src/entities/Publication.ts
Comment (cswenberg) [inline src/entities/Publication.ts:12]: same thing about the Article <-> Publication reference, why is this a string?

File: src/entities/Publication.ts
Comment (cswenberg) [inline src/entities/Publication.ts:21]: if I were to be extra picky, I'd say this should be `websiteURL` to match the style from `Article.ts`

File: src/resolvers/PublicationResolver.ts
Comment (cswenberg) [inline src/resolvers/PublicationResolver.ts:18]: abc?

File: utils/createDummyData.ts
Comment (cswenberg) [inline utils/createDummyData.ts:62]: i'm guessing this can't be abc'd because the shorthand syntax expects the arguments to come in the same order as the fields are defined in the entity file?

**Review by cswenberg — COMMENTED** [review]
I don't want to block this PR to get the caboose moving but could we also get a more populated README (things like setup and such to begin)? Not essential right now, but we should try to build the habit of new documentation for every contribution we make. 

Also non-essential but consider rebasing to clean up your commit history and fix some of the names (some typo and convention things).

File: src/entities/Publication.ts
Comment (cswenberg) [reply src/entities/Publication.ts:12]: also if this is supposed to represent the articles belonging to this publication, wouldn't the name `articles` be more direct and clear? to me, a feed is a more generalistic collection of articles that are pulled together from multiple sources, so in that sense, it's awkward to say this "feed" belongs to this publication vs. these "articles" belong to this publication.

Comment (aliciaxw) [reply src/entities/Publication.ts:12]: agreed

File: utils/createDummyData.ts
Comment (aliciaxw) [inline utils/createDummyData.ts:17]: lol

File: src/entities/Article.ts
Comment (aliciaxw) [inline src/entities/Article.ts:7]: there's been a push to name the id field `uuid` and not `id`

**Review by aliciaxw — CHANGES_REQUESTED** [review]
i would address conner's comment about Article <-> Publication reference in your entities and the minor comments about consistent naming (img vs imageURL), but things like tests and readme can be done in separate prs! in general, amazing work, and really excited to see appdev's first nosql database :-) 🚀

File: src/entities/Article.ts
Comment (orkosinha) [reply src/entities/Article.ts:15]: Not with the current design because Publication isn’t a table being stored in the database. However, I can change this to being a MongoDB entity and adding a new table for publishers. The motivation behind the design was that publisher data is static, so there wasn’t too much of a need to make a table for it. This isn’t really scalable though for more features in the future so it may be worth changing.

File: src/entities/Publication.ts
Comment (orkosinha) [reply src/entities/Publication.ts:15]: Fixed

File: src/entities/Article.ts
Comment (aliciaxw) [reply src/entities/Article.ts:15]: @orkosinha let's keep what you have so far and make a github issue for it to change it after this PR is merged tbh, as long as we're aware there's a more scalable solution

File: src/db/rss-parser.ts
Comment (orkosinha) [reply src/db/rss-parser.ts]: fixed

File: src/entities/Publication.ts
Comment (orkosinha) [reply src/entities/Publication.ts]: fixed, mostly was an issue with `rss-parser.ts` but I think it reads better now

File: src/entities/Publication.ts
Comment (orkosinha) [reply src/entities/Publication.ts]: Agreed, fixed

File: src/resolvers/PublicationResolver.ts
Comment (orkosinha) [reply src/resolvers/PublicationResolver.ts]: Fixed

File: utils/createDummyData.ts
Comment (orkosinha) [reply utils/createDummyData.ts:62]: ArticleModel is never using create like this in our code, so it isn't an issue.

File: src/entities/Article.ts
Comment (orkosinha) [reply src/entities/Article.ts]: Fixed

**Review by aliciaxw — APPROVED** [review]
great edits, LGTM!!! go volume!!!!!!

**Pull request merged**
